### PR TITLE
fix(scripts/reorder-search-index): no null values

### PR DESCRIPTION
**Pull request by caugner**
## Summary

(MP-1795)

### Problem

The `de` search index contained `null` values for two reasons:

1. The script ran twice during `stage-build`.
2. The script assumed that every page in `en-US` is already translated in `de`.

### Solution

1. Remove the duplicate script call.
2. ~Use the `en-US` search index entry for pages that aren't translated yet.~

---

## Screenshots

(No visual change.)

---

## How did you test this change?

Downloaded the `en-US` and `de` search indexes from both prod and stage, and ran the script on them.

### CODE REVIEW

File: scripts/reorder-search-index.mjs [hunk 1]
```diff
@@ -7,19 +7,29 @@ async function main() {
...
+  const inputSlugs = input.map(getSlug);
 
   const result = [];
-  for (const [fromIndex, toIndex] of input
-    .map(getSlug)
-    .map(getIndex)
-    .entries()) {
-    result[toIndex] = input[fromIndex];
+
+  // Add all reference items that are in the reference.
```
Comment (caugner) [inline scripts/reorder-search-index.mjs:19]:
```suggestion
  // Add en entry for each reference item.
```

File: scripts/reorder-search-index.mjs [hunk 2]
```diff
@@ -7,19 +7,29 @@ async function main() {
...
+  for (const [refIndex, slug] of refSlugs.entries()) {
+    const inputIndex = inputSlugs.indexOf(slug);
+    // Use reference item where index does not have this item.
```
Comment (caugner) [inline scripts/reorder-search-index.mjs:22]:
```suggestion
    // Use reference item where it's currently missing in index.
```

File: scripts/reorder-search-index.mjs [hunk 3]
```diff
@@ -7,19 +7,29 @@ async function main() {
...
+    const item = inputIndex !== -1 ? input[inputIndex] : ref[refIndex];
+    result.push(item);
+  }
+
+  // Add items that are NOT in the reference (e.g. moved/removed).
```
Comment (caugner) [inline scripts/reorder-search-index.mjs:27]:
```suggestion
  // Add entries for items that are NOT in the reference (e.g. moved/removed).
```

**Comment by fiji-flo** [comment]
We can revisit this, but for now I backed out the en-US fallback and changed the script to avoid the loop over `includes` and `indexOf` which makes this run in $O(n^2)$.

**Pull request merged**